doc/api/next_api_changes/behavior/26901-RP.rst

```diff
@@ -0,0 +1,4 @@
+``set_xdata`` and ``set_ydata`` requires an iterable.
```

Comment:
This should mention `Line2D`.  Feel free to copy from the original deprecation release note:

https://github.com/matplotlib/matplotlib/blob/602803912e7294d1882378b5601d19f5a5e7aebc/doc/api/prev_api_changes/api_changes_3.7.0/deprecations.rst?plain=1#L287-L291